pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -1812,6 +1812,15 @@ func (wf *Workflow) GetTemplateByName(name string) *Template {
 	return nil
 }
 
+func (wf *Workflow) GetNodeByName(nodeName string) *NodeStatus {
+	nodeID := wf.NodeID(nodeName)
+	node, ok := wf.Status.Nodes[nodeID]
+	if !ok {
+		return nil
+	}
+	return &node
+}
```

Comment:
This function is just ported from the controller, this is not new code.

Reply:
can we call `FindNodeByName` as it return nil?

Reply:
Every instance where this is used there is a nil check done